### PR TITLE
Update task taxonomy

### LINKED ISSUE #10825: create new scope  in `Task` that finds actionable tasks  that filters out `on_hold` tasks
create a scope in `Task` that finds actionable tasks - this would entail renaming `active` to `open` and use `active` to mean actionable

- update all usages/instances of `tasks.active.where.not(status: :on_hold)` to use this new scope

- update `most_recently_assigned_to_label` in `app/models/appeal.rb ` to use this new scope

- fix the 30+ failing tests as a result of this change

### PR DESCRIPTION
**Pull request by youngfreezyVA**
Resolves #10825

### CODE REVIEW

File: app/models/queues/attorney_queue.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ class AttorneyQueue
   # using assigned by user. We set status to being on_hold and placed_on_hold_at to assigned_at timestamp
   def tasks
     colocated_tasks_grouped = ColocatedTask.includes(*task_includes)
-      .active.where(assigned_by: user, assigned_to: Colocated.singleton, appeal_type: LegacyAppeal.name)
+      .open.where(assigned_by: user, assigned_to: Colocated.singleton, appeal_type: LegacyAppeal.name)
```
Comment (youngfreezyVA) [inline app/models/queues/attorney_queue.rb]: @laurjpeterson this should resolve the bug you said was reported ... feel free to link to this PR.

Comment (lowellrex) [reply app/models/queues/attorney_queue.rb]: Does this change alter behaviour in any way? Isn't old "active" the same as new "open"?

File: app/models/appeal.rb [hunk 2]
```diff
@@ -679,7 +679,7 @@ def available_review_options
   private
 
   def most_recently_assigned_to_label(tasks)
-    tasks.order(:updated_at).last.assigned_to_label
+    tasks.order(:created_at).last.assigned_to_label
```
Comment (lowellrex) [inline app/models/appeal.rb]: I think this method needs to be updated to something like `tasks.active.order(:updated_at).last.assigned_to_label` to satisfy the acceptance criteria of this ticket, right?

File: app/models/queues/attorney_queue.rb (hunk 1 above)
Comment (laurjpeterson) [reply app/models/queues/attorney_queue.rb]: i think this is it. @youngfreezyVA could you take a look at it and confirm? https://github.com/department-of-veterans-affairs/appeals-support/issues/5293 

cc: @msj0nes

**Comment by youngfreezyVA** [comment]
@lowellrex @aroltsch tests are passing now, please have a look. @laurjpeterson decided to pull the attorney queue bug into another ticket/PR. when I made the relevant change a lot of tests were failing and I need to investigate further.  https://github.com/department-of-veterans-affairs/caseflow/issues/10968

File: app/models/appeal.rb [hunk 3]
```diff
@@ -684,7 +685,7 @@ def assign_ro_and_update_ahls(new_ro)
   private
 
   def most_recently_assigned_to_label(tasks)
-    tasks.order(:updated_at).last.assigned_to_label
+    tasks.order(:created_at).last&.assigned_to_label
```
Comment (lowellrex) [inline app/models/appeal.rb]: I thought this was the change we were delaying until a later PR?

File: app/models/task.rb [hunk 4]
```diff
@@ -52,18 +54,26 @@ def label
...
     [Constants.TASK_STATUSES.completed, Constants.TASK_STATUSES.cancelled]
   end
 
+  def self.active_statuses
+    [Constants.TASK_STATUSES.assigned, Constants.TASK_STATUSES.in_progress]
+  end
+
+  def self.open_statuses
+    active_statuses.concat([Constants.TASK_STATUSES.on_hold])
+  end
```
Comment (lowellrex) [inline app/models/task.rb]: Can we get rid of this method and replace the one place it is used by defining that scope as the inverse of closed?

```ruby
scope :open, -> { where.not(status: closed_statuses) }
```

**Pull request merged**